test/lseq.rs

```diff
@@ -42,37 +72,20 @@ fn test_delete_of_index() {
 }
 
 #[test]
-fn test_inserts() {
-    // A simple smoke test to ensureethat insertions work properly.
-    // Uses two sites which randomly insert a character and then immediately insert it into the
-    // other site.
-    let mut rng = rand::thread_rng();
-
-    let mut s1 = rng.sample_iter(Alphanumeric);
-    let mut s2 = rng.sample_iter(Alphanumeric);
-
-    let mut site1 = LSeq::new(SiteId::new(0));
-    let mut site2 = LSeq::new(SiteId::new(1));
-
-    for _ in 0..5000 {
-        if rng.gen() {
-            let op = site1.insert_index(
-                rng.gen_range(0, site1.raw_entries().len() + 1),
-                s1.next().unwrap(),
-            );
-            site2.apply(op);
-        } else {
-            let op = site2.insert_index(
-                rng.gen_range(0, site2.raw_entries().len() + 1),
-                s2.next().unwrap(),
-            );
-            site1.apply(op);
-        }
+fn test_worst_case_inserts() {
```

Comment:
We need to tune some of the allocation behavior, in our worst case, we can't insert more than 86 entries.

Reply:
It should actually be lower than that. However, it's statistically very unlikely (you'd need a whole ot of _very_ bad dice rolls). The only solution is to use `bignum` instead of u64.

Reply:
I don't think the boundary strategies need any changes honestly. I don't think that the pathological behavior that is shown by arbitrary is relevant to the real-world. But if we do want to change it i'd say the change we should actually make is to switch to using `bignum` inside identifiers instead of `u64`.

Reply:
ok, makes sense, lets leave that for a future improvement, it could very likely not be a problem in practice.

Reply:
I've created documents with tens of thousands of entries without any difficulty before so I don't think it manifests frequently.